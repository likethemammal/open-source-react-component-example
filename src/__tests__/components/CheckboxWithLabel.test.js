import React from 'react';
import {shallow} from 'enzyme';

import toJSON from 'enzyme-to-json'


import CheckboxWithLabel from '../../components/CheckboxWithLabel';

describe('CheckboxWithLabel', () => {
    // Render a checkbox with label in the document
    const component = shallow(
        <CheckboxWithLabel labelOn="On" labelOff="Off" />
    )

    test(`Snapshot should match`, () => {
        expect(toJSON(component)).toMatchSnapshot()
    })

    test('changes the text after click', () => {

        expect(component.text()).toEqual('Off');

        component.find('input').simulate('change');

        expect(component.text()).toEqual('On');
    })
})