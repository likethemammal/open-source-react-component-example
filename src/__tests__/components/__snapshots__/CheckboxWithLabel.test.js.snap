// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxWithLabel Snapshot should match 1`] = `
<label>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  Off
</label>
`;
